src/transformers/models/llama/modeling_llama.py

```diff
@@ -1050,6 +1053,10 @@ def forward(
             attentions=all_self_attns,
         )
 
+    # TODO: As of 20/02/2024, the `attention_mask` passed to the model in `generate` is 2D and of dynamic length even when the static
+    # KV cache is used. This is an issue for torch.compile which then recaptures cudagraphs at each decode steps due to the dynamic shapes.
+    # (`recording cudagraph tree for symint key 13`, etc.). Ideally, we would want to pass a statically-shaped 4D mask as input to the model.
+    @torch.compiler.disable
```

Comment:
this slows down a tad bit the forward of sdpa with padding I would believe? I'll check on my benchmarks anyways!

Reply:
There is a slight slowdown indeed, it may be due to this.